src/drivers/amd/renoir/acp_dmic_dai.c

```diff
@@ -46,7 +68,26 @@ static int acp_dmic_dai_trigger(struct dai *dai, int cmd, int direction)
 
 static int acp_dmic_dai_probe(struct dai *dai)
 {
-	/* TODO */
+	struct acp_pdata *acp;
+
+	dai_info(dai, "ACP: acp_dmic_dai_probe");
+	/* allocate private data */
+	acp = rzalloc(SOF_MEM_ZONE_RUNTIME_SHARED, 0, SOF_MEM_CAPS_RAM, sizeof(*acp));
+	if (!acp) {
+		dai_err(dai, "acp_dmic_dai_probe(): alloc failed");
+		return -ENOMEM;
+	}
+	dai_set_drvdata(dai, acp);
+	return 0;
+}
+
+static int acp_dmic_dai_remove(struct dai *dai)
+{
+	struct acp_pdata *acp = dai_get_drvdata(dai);
+
+	dai_info(dai, "acp_dmic_dai_remove()");
+	rfree(acp);
+	dai_set_drvdata(dai, NULL);
```

Comment:
I'd switch the set_drvdata to NULL and rfree

Reply:
Intel and other platforms also doing in same way.
For reference:
https://github.com/thesofproject/sof/blob/main/src/drivers/imx/esai.c#L372
https://github.com/thesofproject/sof/blob/main/src/drivers/intel/ssp/ssp.c#L1147